pytedee/pytedee/TedeeClient.py

```diff
@@ -194,51 +159,49 @@ def get_state(self):
                     if id == lock.get_id():
                         lock.set_connected(is_connected)
                         lockProperties = state["lockProperties"]
-                        _LOGGER.debug("Id: %s, State: %d, battery: %d", state["id"], lockProperties ["state"], lockProperties["batteryLevel"])
-                        battery_level = lockProperties["batteryLevel"]
-                        lock.set_battery_level(battery_level)
+                        lock_state = self.assign_null_or_lock_state(lockProperties)
+                        lock_batteryLevel = self.assign_null_or_lock_state(lockProperties)
+                        lock_isCharging = self.assign_null_or_lock_isCharging(lockProperties)
+                        _LOGGER.debug("Id: %s, State: %d, battery: %d", lock_state, lock_isCharging, lock_batteryLevel)
+
+                        lock.set_battery_level(lock_batteryLevel)
                         '''Todo: do something with state'''
-                        lock.set_state(lockProperties["state"])
+                        lock.set_state(lock_state)
                         break
         except KeyError:
             _LOGGER.error("result: %s", r.json())
             
-    def is_token_valid(self):
-        return self._token_valid_until > datetime.datetime.now()
-
-    def ensure_token_is_valid(self):
-        if (self.is_token_valid() == False):
-            self.get_token()
-        return self.is_token_valid() == True
-
     def find_lock(self, id):
         for lock in self._sensor_list:
             if id == lock.get_id():
                 return lock
         raise TedeeClientException("This Id not found")
-                      
-    def update(self, id):
```

Comment:
For periodically update of the home assistant custom component, function update need to be there:
```
def update(self, id)
  self.get_state()
  return self.get_battery(id)
```